apps/auctions/volume_matching.py

```diff
@@ -0,0 +1,340 @@
+# Original code was written by Harjasleen Malvai
+
+"""
+In volume matching auction, buy and sell orders are matched only on volume
+while price is determined by reference to some external market.
+"""
+
+import asyncio
+import logging
+
+from honeybadgermpc.mpc import TaskProgramRunner
+from honeybadgermpc.preprocessing import (
+    PreProcessedElements as FakePreProcessedElements,
+)
+from honeybadgermpc.progs.fixedpoint import FixedPoint
+from honeybadgermpc.progs.mixins.share_arithmetic import (
+    BeaverMultiply,
+    BeaverMultiplyArrays,
+    MixinConstants,
+)
+
+config = {
+    MixinConstants.MultiplyShareArray: BeaverMultiplyArrays(),
+    MixinConstants.MultiplyShare: BeaverMultiply(),
+}
+
+
+async def compute_bids(ctx, balances, bids, price):
+    """Compute all valid bids for each user.
+    According to current market price, a bid becomes invalid
+    if the user doesn't have enough balance to pay for this bid.
+    We only keep valid bids and classify them into
+    buy bids(volume > 0) and sell bids(volume < 0).
+    Invalid bids will be discarded after executing this function.
+
+    Parameters
+    ----------
+    balances : {address -> {cointype -> balance}}
```

Comment:
for the type, I think the convention is usually to specify the type, such as `list`, `dict`, `str`, etc

so in this case that would be `dict of dict`

in the description, details about the keys and values can be added, such as keys are addresses, and values are dictionaries in which the keys are coin types and the values are balances ...

Just as a rough example ...